src/ManagedCodecs/TargaCodec.cs

```diff
@@ -0,0 +1,131 @@
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+
+using PhotoSauce.MagicScaler;
+
+namespace PhotoSauce.ManagedCodecs
+{
+	/// <summary>Represents a TGA image file.</summary>
+	public sealed class TargaContainer : IImageContainer, IDisposable
+	{
+		private readonly Image decodedImage;
+
+		private TargaContainer(Image image) => decodedImage = image;
+
+		/// <inheritdoc />
+		public FileFormat ContainerFormat => FileFormat.Bmp;
+
+		/// <inheritdoc />
+		public int FrameCount => 1;
+
+		/// <inheritdoc />
+		public IImageFrame GetFrame(int index) => index == 0 ? new TargaFrame(decodedImage) : throw new IndexOutOfRangeException("Invalid frame index");
+
+		/// <summary>Loads a TGA image from the specified path.</summary>
+		/// <param name="imgPath">The path to the image file.</param>
+		/// <returns>A <see cref="TargaContainer"/> encapsulating the image.</returns>
+		public static TargaContainer Load(string imgPath)
+		{
+			int bpp = 0;
+			using (var fs = File.OpenRead(imgPath))
+			{
+				var info = Image.Identify(fs, out var fmt);
+				if (fmt.Name != "TGA")
+					throw new InvalidDataException("Not a TGA image");
+
+				bpp = info.PixelType.BitsPerPixel;
+			}
+
+			return bpp switch
+			{
+				 8 => new TargaContainer(Image.Load<L8>(imgPath)),
+				24 => new TargaContainer(Image.Load<Bgr24>(imgPath)),
+				 _ => new TargaContainer(Image.Load<Bgra32>(imgPath)),
+			};
+		}
+
+		public void Dispose() => decodedImage.Dispose();
+	}
+
+	/// <summary>Gives access to the pixels in a TGA file.</summary>
+	public sealed class TargaFrame : IImageFrame
+	{
+		private readonly Image decodedImage;
+
+		internal TargaFrame(Image image) => decodedImage = image;
+
+		/// <inheritdoc />
+		public double DpiX => 96;
+
+		/// <inheritdoc />
+		public double DpiY => 96;
+
+		/// <inheritdoc />
+		public Orientation ExifOrientation => Orientation.Normal;
+
+		/// <inheritdoc />
+		public ReadOnlySpan<byte> IccProfile => default;
+
+		/// <inheritdoc />
+		public IPixelSource PixelSource => new ImagePixelSource(decodedImage, decodedImage.Frames.RootFrame);
+
+		/// <inheritdoc />
+		public void Dispose() => decodedImage.Dispose();
+	}
+
+	internal sealed class ImagePixelSource : IPixelSource
+	{
+		private readonly Image decodedImage;
+		private readonly ImageFrame decodedFrame;
+
+		public ImagePixelSource(Image image, ImageFrame frame) => (decodedImage, decodedFrame) = (image, frame);
+
+		public Guid Format => decodedFrame switch {
+			ImageFrame<L8>    _ => PixelFormats.Grey8bpp,
+			ImageFrame<Bgr24> _ => PixelFormats.Bgr24bpp,
+			                  _ => PixelFormats.Bgra32bpp
+		};
+
+		public int Width => decodedFrame.Width;
+
+		public int Height => decodedFrame.Height;
+
+		public void CopyPixels(System.Drawing.Rectangle sourceArea, int cbStride, Span<byte> buffer)
+		{
+			var (rx, ry, rw, rh) = (sourceArea.X, sourceArea.Y, sourceArea.Width, sourceArea.Height);
+			int bpp = decodedImage.PixelType.BitsPerPixel / 8;
```

Comment:
Was there a better way to find this info?  It was kind of awkward I had to carry the `Image` with me here to have access to the `PixelType`.  Seems like that should belong to the `ImageFrame` anyway, given some containers (ok, maybe just TIFF) allow a different encoding per frame.

Either way, this would be another place a more generalized descriptor of the `PixelType` would have been useful.

Reply:
> [...] given some containers (ok, maybe just TIFF) allow a different encoding per frame.

This leads to complex architectural questions for us, I don't think we'll be able to resolve them easily, which means that an ImageSharp 1.0 `Image` could be composed only from frames of the same pixel type.

Reply:
@saucecontrol BitsPerPixel is stored in the metadata for the image.

```c#
decodedImage.Metadata.GetTgaMetadata().BitsPerPixel;
```

Reply:
Ah, gotcha.  That makes sense, but I was hoping to get the pixel format info from the frame so I didn't have to carry around the `Image` reference when operating on the `ImageFrame`.  I suppose I can just infer it from `TPixel` as I do further down.